### PR TITLE
SW-4623 Validate accession and species when saving a new batch 

### PR DESCRIPTION
**Pull request by nickgraz**
- Use the new accessions redux search within the useAccessions hook
- Add accession ID and species ID validation to the batch save request flow

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1958?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #1957** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1957?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1958** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1958?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/1958?utm_source=stack-comment).

File: src/components/InventoryV2/form/BatchDetailsForm.tsx [hunk 1]
```diff
@@ -215,6 +219,15 @@ export default function BatchDetailsForm(props: BatchDetailsModalProps): JSX.Ele
 
   const isUndefinedQuantity = (val?: string | number) => val === undefined || val === '';
 
+  // Make sure the selected accession is represented in the record, this will clear
+  // out an "impossible" accession selection
```
Comment (nickgraz) [inline src/components/InventoryV2/form/BatchDetailsForm.tsx:223]: This is the UI validation / scrubbing

File: src/redux/features/batches/batchesAsyncThunks.ts [hunk 2]
```diff
@@ -6,19 +6,38 @@ import { Batch, CreateBatchRequestPayload, NurseryTransfer } from 'src/types/Bat
...
+    { rejectWithValue, getState }
+  ) => {
+    const { batch, organizationId, timezone } = request;
 
     let response: (Response & BatchData) | (Response & BatchId) | undefined;
     let responseQuantities: Partial<Response> = { requestSucceeded: true, error: undefined };
 
     if (batch.id === -1) {
       if (batch.accessionId) {
+        // Extra validation to ensure an accession ID matches the given species
```
Comment (nickgraz) [inline src/redux/features/batches/batchesAsyncThunks.ts:27]: This is the request flow validation

Comment (karthikbtf) [reply src/redux/features/batches/batchesAsyncThunks.ts:27]: Should we be ensuring a valid selection by way of populating correct entries in the form instead?
When a species is selected, the accession dropdown entries should only be relevant to the species.. and vice-versa.

**Review by karthikbtf — COMMENTED** [review]
code looks good, just one question/comment on alternative approach

File: src/redux/features/batches/batchesAsyncThunks.ts (hunk 2 above)
Comment (nickgraz) [reply src/redux/features/batches/batchesAsyncThunks.ts:27]: Yeah, that validation is already in place [here](https://github.com/terraware/terraware-web/pull/1958/files#diff-a4bbdce52d6392f56289439f849e8a49a344471c5da2b2fbbe950ca165159768R222) where is UI scrubbing / validating based on the species selection

**Review by karthikbtf — APPROVED** [review]

File: src/redux/features/batches/batchesAsyncThunks.ts (hunk 2 above)
Comment (karthikbtf) [reply src/redux/features/batches/batchesAsyncThunks.ts:27]: Got it, thanks 👍

**Comment by nickgraz** [comment]
### Merge activity

* **Jan 3, 10:59 AM**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/1958).

**Pull request merged**